### PR TITLE
8343780: Add since checker tests to the Tools area modules and add missing @since to  jdk.jfr

### PR DESCRIPTION
**Pull request by nizarbenalla**
Can I please get a review for this PR that add tests to verify the value of `@since` tags to the Tools area modules. The test is described in this [email](https://mail.openjdk.org/pipermail/jdk-dev/2024-October/009474.html). This issue is similar to JDK-8341399, JDK-8331051 and JDK-8343442.

The benefit from this is helping API authors and reviewer validate the accuracy of `@since` in their source code (and subsequently, in the generated documentation). And lessen the burden on Reviewers.

The test has been added for `java.base` and a few other modules and has helped catch some bugs before they make it to the JDK.

Notes: 
- I have also added an `@since` tag that was missing.
- You will notice there is no test for the `jdk.jfr` module, it will be added in a future PR because it requires special handling. (JFR used to be a commercial feature and this requires special handling to be added for it in the test)

TIA

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8343780](https://bugs.openjdk.org/browse/JDK-8343780): Add since checker tests to the Tools area modules and add missing @since to  jdk.jfr (**Sub-task** - P4)

### Reviewers
 * [Christian Stein](https://openjdk.org/census#cstein) (@sormuras - Committer)
 * [Erik Gahlin](https://openjdk.org/census#egahlin) (@egahlin - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21982/head:pull/21982` \
`$ git checkout pull/21982`

Update a local copy of the PR: \
`$ git checkout pull/21982` \
`$ git pull https://git.openjdk.org/jdk.git pull/21982/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21982`

View PR using the GUI difftool: \
`$ git pr show -t 21982`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21982.diff">https://git.openjdk.org/jdk/pull/21982.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21982#issuecomment-2465070351)
</details>

### CODE REVIEW

File: test/jdk/tools/sincechecker/modules/jdk.jlink/JdkJlinkCheckSince.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8343780
+ * @summary Test for `@since` in jdk.jlink module
+ * @library /test/lib /test/jdk/tools/sincechecker
+ * @run main SinceChecker jdk.jlink
```
Comment (sormuras) [inline test/jdk/tools/sincechecker/modules/jdk.jlink/JdkJlinkCheckSince.java:29]: Missing "last" line? 
```suggestion
 * @run main SinceChecker jdk.jlink
 */
```

**Review by sormuras — CHANGES_REQUESTED** [review]
Two nits:
- Copyright notice in `Recording.java` needs a 2024
- A missing line in a test file

File: test/jdk/tools/sincechecker/modules/jdk.jlink/JdkJlinkCheckSince.java (hunk 1 above)
Comment (nizarbenalla) [reply test/jdk/tools/sincechecker/modules/jdk.jlink/JdkJlinkCheckSince.java:29]: You're right. Will update, thanks.

Comment (nizarbenalla) [reply test/jdk/tools/sincechecker/modules/jdk.jlink/JdkJlinkCheckSince.java:29]: The copyright year was fixed after a merge :)

**Review by sormuras — APPROVED** [review]
Looks good to me.

**Review by egahlin — APPROVED** [review]
JFR change looks good.

**Comment by nizarbenalla** [comment]
Added one last trivial change, an \@since 16 to `RecordingStream.onMetadata` which was added in [JDK-8248564](https://bugs.openjdk.org/browse/JDK-8248564). `{@inheritdoc}` is implicit and is not needed.

**Comment by nizarbenalla** [comment]
Reverting the change to `RecordingStream`, as [JDK-8345339](https://bugs.openjdk.org/browse/JDK-8345339) will filed.

**Pull request closed without merging**